lib/spack/spack/cmd/tags.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+import six
+
+import llnl.util.tty.colify as colify
+
+import spack.repo
+import spack.tag
+
+description = "list package tags and or their packages"
+section = "basic"
+level = "long"
+
+
+def report_tags(category, tags):
+    buffer = six.StringIO()
+    isatty = sys.stdout.isatty()
+
+    if isatty:
+        buffer.write("{0} package tags:\n".format(category))
+
+    if tags:
+        colify.colify(tags, output=buffer, tty=isatty, indent=4)
+    else:
+        buffer.write("    None\n")
+    print(buffer.getvalue())
+
+
+def setup_parser(subparser):
+    subparser.epilog = "Processes available tags when called without tags."
+    subparser.add_argument(
+        '-i', '--installed', action='store_true', default=False,
+        help="limit tags to those of installed packages"
+    )
+    subparser.add_argument(
+        '-s', '--show-packages', action='store_true', default=False,
+        help="show packages for available tags when no tags are provided"
+    )
```

Comment:
What do you think  about calling this:
```suggestion
    subparser.add_argument(
        '--all', action='store_true', default=False,
        help="show packages for all the available tags"
    )
```
and making it such that either this option or a list of tags can be provided from the command line?

Reply:
IIRC I was having sub parser problems with mutual exclusion but will give it another try.

Reply:
I don't appear (Python 3.7?) to be able to have the packages option mutually exclusive with the positional tag arguments *unless* I'm willing to support only a single tag and that was not the intention.

Reply:
https://github.com/spack/spack/blob/361d98c4ec93f9b682b949a111958df38e70f46b/lib/spack/spack/cmd/tags.py#L61-L64 

This is absolutely fine, thanks!